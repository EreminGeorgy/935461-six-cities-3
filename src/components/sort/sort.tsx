import * as React from "react";
import {useState} from "react";
import PropTypes from "prop-types";

const MENU_STATE = {
  OPENED: `opened`,
  CLOSED: `closed`
};

export const Sort = (props) => {
  const {setItem} = props;
  const [closed, toggleState] = useState(true);
  const [text, setText] = useState(`Popular`);

  const handleTabClick = (target) => {
    setItem(target);
    setText(target.textContent);
    toggleState(true);
  };

  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span onClick={() => toggleState(!closed)} className="places__sorting-type" tabIndex="0">
        {text}
        <svg className="places__sorting-arrow" width="7" height="4">
          <use xlinkHref="#icon-arrow-select"/>
        </svg>
      </span>
      <ul className={`places__options places__options--custom places__options--${closed ? MENU_STATE.CLOSED : MENU_STATE.OPENED}`}>
        <li className="places__option" id="default" onClick={(evt) => {
          handleTabClick(evt.target);
        }} tabIndex="0">Popular</li>
        <li className="places__option" id="lowToHigh" onClick={(evt) => {
          handleTabClick(evt.target);
        }} tabIndex="0">Price: low to high</li>
        <li className="places__option" id="highToLow" onClick={(evt) => {
          handleTabClick(evt.target);
        }} tabIndex="0">Price: high to low</li>
        <li className="places__option" id="topRated" onClick={(evt) => {
          handleTabClick(evt.target);
        }} tabIndex="0">Top rated first</li>
      </ul>
    </form>
  );
};

Sort.propTypes = {
  setItem: PropTypes.func,
};
