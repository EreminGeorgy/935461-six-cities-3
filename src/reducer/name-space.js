const NameSpace = {
  DATA: `DATA`,
  APPLICATION: `APPLICATION`,
  USER: `USER`,
  COMMENTS: `COMMENTS`,
  FAVORITES: `FAVORITES`,
};

export default NameSpace;
