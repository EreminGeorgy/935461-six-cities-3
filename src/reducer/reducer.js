import {offers} from '../mocks/offers';
import {extend, getCities, getOffersByCity} from '../utils/utils.js';

const citiesList = getCities(offers);
const initialOffers = getOffersByCity(offers, citiesList[0]);

const initialState = {
  activeCity: citiesList[0],
  offersInActiveCity: initialOffers,
  cities: citiesList,
};

const ActionCreator = {
  newCity: (newCity) => ({
    type: `NEW_CITY`,
    payload: newCity,
  }),
  getOffers: (newCity) => ({
    type: `GET_OFFERS`,
    payload: newCity,
  }),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `NEW_CITY`:
      return extend(state, {
        activeCity: action.payload,
      });
    case `GET_OFFERS`:
      return extend(state, {
        offersInActiveCity: getOffersByCity(offers, action.payload),
      });
  }
  return state;
};

export {reducer, ActionCreator};
