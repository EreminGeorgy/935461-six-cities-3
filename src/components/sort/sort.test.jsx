import React from "react";
import renderer from "react-test-renderer";
import {Sort} from "./sort.jsx";
import {Provider} from 'react-redux';
import reducer from '../../reducer/reducer.js';
import {createStore} from 'redux';

const store = createStore(
    reducer
);


it(`<Sort /> should render Sort`, () => {
  const tree = renderer
    .create(<Provider store={store}>
      <Sort
        setItem={()=>{}}
      />
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
