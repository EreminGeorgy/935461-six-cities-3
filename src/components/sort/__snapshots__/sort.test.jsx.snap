// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sort /> should render Sort 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--closed"
  >
    <li
      className="places__option"
      id="default"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      id="lowToHigh"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      id="highToLow"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      id="topRated"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
